src/components/App.tsx

```diff
@@ -1350,21 +1347,27 @@ class App extends React.Component<AppProps, AppState> {
         scale: window.devicePixelRatio,
         rc: this.rc!,
         canvas: this.canvas!,
-        renderConfig: {
-          selectionColor,
+        canvasUi: this.canvasUi!,
+        canvasContentRenderConfig: {
```

Comment:
instead of constructing the objects here (and duplicating some of the props), let's construct them inside `renderScene()`?

Reply:
then we need to create the type `RenderConfig` as before, and inside `_renderScene()` we have to distribute it into two, which doesn't make sense or of any use,

Reply:
> then we need to create the type RenderConfig as before

Yeah, I think it's fine. We'd have `RenderConfig` and then two derived types for each of the sub canvas functions.

But let's leave it for now then.